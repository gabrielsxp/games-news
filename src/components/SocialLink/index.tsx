import * as S from './styles'

const SocialLink = () => (
  <S.Wrapper>
    <h1>SocialLink</h1>
  </S.Wrapper>
)

export default SocialLink
