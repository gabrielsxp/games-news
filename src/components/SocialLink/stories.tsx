import { Story, Meta } from '@storybook/react/types-6-0'
import SocialLink from '.'

export default {
  title: 'SocialLink',
  component: SocialLink
} as Meta

export const Default: Story = () => <SocialLink />
