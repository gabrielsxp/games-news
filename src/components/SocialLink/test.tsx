import { render, screen } from '@testing-library/react'

import SocialLink from '.'

describe('<SocialLink />', () => {
  it('should render the heading', () => {
    const { container } = render(<SocialLink />)

    expect(screen.getByRole('heading', { name: /SocialLink/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
