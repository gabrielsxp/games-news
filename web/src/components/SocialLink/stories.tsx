import { Story, Meta } from '@storybook/react/types-6-0'
import SocialLink, { SocialLinkProps } from '.'
import mock from './mock'

export default {
  title: 'SocialLink',
  component: SocialLink,
  argTypes: {
    color: { control: 'color' }
  }
} as Meta

export const Default: Story<SocialLinkProps> = (args) => (
  <SocialLink {...args} />
)

Default.args = {
  ...mock,
  title: 'Facebook'
}

Default.parameters = {
  layout: 'centered'
}
