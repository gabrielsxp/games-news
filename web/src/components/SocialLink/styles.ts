import styled, { css } from 'styled-components'
import { lighten } from 'polished'
import { SocialBannerProps } from 'components/SocialBanner'

export const Wrapper = styled.a<Pick<SocialBannerProps, 'color'>>`
  ${({ theme, color }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    width: 4.5rem;
    height: 4.5rem;
    border-radius: ${theme.border.radius};
    background-color: ${lighten(0.2, theme.colors.secondary)};
    position: relative;
    transition: background-color 0.3s ease;

    &:hover {
      background-color: ${color};
    }
  `}
`

export const IconWrapper = styled.div`
  ${({ theme }) => css`
    width: 2rem;
    height: 2rem;
    > svg {
      width: 100%;
      height: 100%;
      color: ${theme.colors.white};
    }
  `}
`
