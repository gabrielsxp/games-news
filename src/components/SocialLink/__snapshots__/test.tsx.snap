// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SocialLink /> should render the heading 1`] = `
<main
  class="p997hr-0"
>
  <h1>
    SocialLink
  </h1>
</main>
`;
